plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1692,6 +1717,24 @@ private void checkUnsupportedGeneratedColumns(MetadataEntry metadataEntry)
         }
     }
 
+    private void checkUnsupportedColumnMapping(MetadataEntry metadataEntry)
+    {
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry);
+        if (!(columnMappingMode == ColumnMappingMode.NONE || columnMappingMode == ColumnMappingMode.NAME)) {
+            throw new TrinoException(NOT_SUPPORTED, "Writing with column mapping %s is not supported".formatted(columnMappingMode.name().toLowerCase(ENGLISH)));
```

Comment:
I think the phrase 'Writing with' in the error message could be omitted, as `checkUnsupportedColumnMapping` does not indicate whether it is being called for DML. It could also be called for DDL operations in the future.

Reply:
Agreed. OTOH, saying "Column mapping ID is not supported" would be misleading, since it's partially supported (supported for reads).

Let's keep current wording and rename the method to indicate it should be called in context of writes only.